FWCore/Framework/src/Worker.h

```diff
@@ -984,9 +980,12 @@ namespace edm {
         auto ownRunTask = std::make_shared<DestroyTask>(runTask);
         auto selectionTask = make_waiting_task(
             tbb::task::allocate_root(),
-            [ownRunTask, parentContext, info = transitionInfo, token, this](std::exception_ptr const*) mutable {
-              ServiceRegistry::Operate guard(token);
-              prefetchAsync<T>(ownRunTask->release(), token, parentContext, info, T::transition_);
+            [ownRunTask, parentContext, info = transitionInfo, token, this](std::exception_ptr const* iExcept) mutable {
+              //NOTE: a non null value of iExcept means the selection succeeded
+              if (!iExcept) {
```

Comment:
Could you elaborate? Was there a bug before?

Reply:
That is part of a change I tried to back out. Evidently I missed a piece.